### PR TITLE
[MediaController] Add new APIs for event, capabilities and search.

### PR DESCRIPTION
**Pull request by hsgwon**
### Description of Change ###

Add new APIs for event and capabilities, search.

### API Changes ###

 - TCSACR-186

### CODE REVIEW

**Comment by WonyoungChoi** [comment]
Run Build Checker

File: src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs [hunk 1]
```diff
@@ -28,17 +29,23 @@ namespace Tizen.Multimedia.Remoting
     /// <since_tizen> 5 </since_tizen>
     public abstract class Command
     {
-        private string _requestId;
+        /// <summary>
+        /// The request id for each command.
+        /// </summary>
+        protected string _requestId;
```
Comment (WonyoungChoi) [inline src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs]: The protected variable is also be publicly exported. So It should be included in ACR scope. 
And the undersocre prefix should be used for private variables only.

File: src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs [hunk 2]
```diff
@@ -334,16 +343,181 @@ internal override string Request()
...
+
+            if (_searchHandle != IntPtr.Zero)
+            {
+                NativeClient.DestroySearchHandle(_searchHandle).ThrowIfError("Failed to destroy search handle");
+            }
+
+            return requestId;
+        }
+
+        internal override void Response(IntPtr handle, int result, Bundle bundle)
```
Comment (myroot) [inline src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs]: It is not good design, It is bad to OCP.
SubClass of Command need not to know detail of Response implementation

I recommand add a protected method `OnResponseCompleted` on Command class
``` c#
class abstract class Command {
...
 protected virtual OnResponseCompleted() {}
 internal void Response(IntPtr handle, int result, Bundle bundle)
 {
     try {
         if (bundle != null)
         {
             NativeServer.SendCommandReplyBundle(handle, _clientId, _requestId, result, bundle.SafeBundleHandle)
                 .ThrowIfError("Failed to response command.");
          }
          else
          {
              NativeServer.SendCommandReply(handle, _clientId, _requestId, result, IntPtr.Zero)
                  .ThrowIfError("Failed to response command.");
           }
      }
      finally
      {
           OnResponseCompleted();
      }
 }
...
}
```
And override `OnResponseCompleted` in SearchCommand class
``` c#
class SearchCommand {
  ...
 protected override OnResponseCompleted() {
   if (_searchHandle != IntPtr.Zero)
   {
        NativeClient.DestroySearchHandle(_searchHandle).ThrowIfError("Failed to destroy search handle");
    }
 }
  ...
}
```

File: src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs [hunk 3]
```diff
@@ -334,16 +343,181 @@ internal override string Request()
...
+                            condition.Keyword, IntPtr.Zero).
+                            ThrowIfError("Failed to set search condition.");
+                    }
+                }
+            }
+            catch
+            {
+                if (_searchHandle != IntPtr.Zero)
+                {
+                    NativeClient.DestroySearchHandle(_searchHandle).ThrowIfError("Failed to destroy search handle");
```
Comment (myroot) [inline src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs]: Why not rethrow exception?

File: src/Tizen.Multimedia.Remoting/MediaController/MediaControlSearchCondition.cs [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        /// </summary>
+        /// <remarks><see cref="MediaControlSearchCategory"/> will be set 'ALL'.</remarks>
+        /// <since_tizen> 5 </since_tizen>
+        public MediaControlSearchCondition(MediaControlContentType type, string keyword, Bundle bundle)
+        {
+            ValidationUtil.ValidateEnum(typeof(MediaControlContentType), type, nameof(type));
+
+            ContentType = type;
+            Keyword = keyword ?? throw new ArgumentNullException(nameof(keyword));
+            Bundle = bundle ?? null;
```
[the remark this answers is not in the document]
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/MediaController/MediaControlSearchCondition.cs]: It's mistake. I'll fix it.

File: src/Tizen.Multimedia.Remoting/MediaController/MediaControlSearchCondition.cs [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+        /// Gets the search content type.
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        public MediaControlContentType ContentType { get; }
+
+        /// <summary>
+        /// Gets the search category.
+        /// </summary>
+        /// <since_tizen> 5 </since_tizen>
+        public MediaControlSearchCategory Category { get; } = 0;
```
[the remark this answers is not in the document]
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/MediaController/MediaControlSearchCondition.cs]: In native fw, it' means 'search it in all category'. I'll add it as "All'.

File: src/Tizen.Multimedia.Remoting/MediaController/MediaController.cs [hunk 6]
```diff
@@ -194,7 +196,48 @@ public IEnumerable<MediaControlPlaylist> GetPlaylists()
...
+            }
+            finally
+            {
+                if (playbackHandle != IntPtr.Zero)
+                {
+                    Native.DestroyPlayback(playbackHandle).ThrowIfError("Failed to destroy playback handle.");
+                }
+            }
+
+            var playlist = from list in GetPlaylists()
```
[the remark this answers is not in the document]
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/MediaController/MediaController.cs]: I'll change it.

File: src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs (hunk 2 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs]: Ok. I'll change it.

File: src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs (hunk 3 above)
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/MediaController/MediaControlCommand.cs]: It's mistake. I'll fix it.

File: src/Tizen.Multimedia.Remoting/MediaController/MediaControlServer.cs [hunk 7]
```diff
@@ -311,6 +314,77 @@ internal static IntPtr GetPlaylistHandle(string name)
...
+        /// </exception>
+        /// <returns>the activated client ids.</returns>
+        /// <since_tizen> 5 </since_tizen>
+        public static IEnumerable<string> GetActivatedClients()
+        {
+            List<string> clientIds = null;
+
+            Native.ActivatedClientCallback activatedClientCallback = (name, _) =>
+            {
+                clientIds.Add(name);
```
[the remark this answers is not in the document]
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/MediaController/MediaControlServer.cs]: It's set by ```appid``` in client creation time. Server also knows it already, not runtime.

[the remark this answers is not in the document]
Comment (myroot) [reply src/Tizen.Multimedia.Remoting/MediaController/MediaControlServer.cs]: I mean, who is create `List<string>`
``` c#
List<string> clientIds = null;
```

[the remark this answers is not in the document]
Comment (hsgwon) [reply src/Tizen.Multimedia.Remoting/MediaController/MediaControlServer.cs]: I'll fix it.

**Comment by myroot** [comment]
OMG why it merged?

**Comment by hsgwon** [comment]
Wonyoung choi was approved it. Is there any problem?

**Pull request merged**